File: problem_set/check-balance-lcci/check-balance-lcci.cpp

```
// @Title: 检查平衡性 (Check Balance LCCI)
// @Author: robert.sunq
// @Date: 2020-04-09 20:48:17
// @Runtime: 12 ms
// @Memory: 20.5 MB

/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
protected:
    int isbalan(TreeNode * root){
        if(root == NULL) return 0;
        else{
            
            
            int deep_left = isbalan(root->left) +1;
            
            int deep_right =isbalan(root->right)+1;
            if(deep_left <0 || deep_right <0) return -5;
            if(abs(deep_left-deep_right) >1) return -5;
            return deep_left > deep_right ? deep_left :deep_right;
        }
    }
public:
    bool isBalanced(TreeNode* root) {
        if(root == NULL) return true;
        
        int deep = isbalan(root);
        if(deep < 0) return false;
        else return true;

    }
};

File: problem_set/check-balance-lcci/check-balance-lcci.java

```
// @Title: 检查平衡性 (Check Balance LCCI)
// @Author: robert.sunq
// @Date: 2021-08-02 23:19:44
// @Runtime: 1 ms
// @Memory: 38.6 MB

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public boolean isBalanced(TreeNode root) {
        return recur(root) >-1;
    }
    public int recur(TreeNode root){
        if(root == null) return 0;
        int left = recur(root.left);
        if(left == -1) return -1;
        int right = recur(root.right);
        if(right == -1) return -1;
        return Math.abs(left - right) >1 ? -1:Math.max(left , right) +1;
    }
}
